kart/point_cloud/import_.py

```diff
@@ -250,12 +247,30 @@ def get_actual_merged_metadata(self, all_metadata):
         )
 
     def get_conversion_func(self, tile_source):
-        if self.convert_to_cloud_optimized and not is_copc(tile_source.metadata):
+        if self.override_crs:
+            # When override_crs is specified, we always need to convert
+            if self.convert_to_cloud_optimized:
+                # Convert to COPC (or maintain COPC) with CRS override
+                return lambda source, dest: convert_tile_to_copc(
+                    source, dest, override_srs=self.override_crs
+                )
+            else:
+                # Convert with CRS override, preserving original format
+                return lambda source, dest: convert_tile_with_crs_override(
+                    source, dest, override_srs=self.override_crs
+                )
+        elif self.convert_to_cloud_optimized and not is_copc(tile_source.metadata):
+            # Convert to COPC without CRS override
             return convert_tile_to_copc
         return None
 
     def existing_tile_matches_source(self, source_oid, existing_summary):
         """Check if the existing tile can be reused instead of reimporting."""
+        # If override_crs is specified, we need to rewrite the CRS in the files,
+        # so we can't reuse existing tiles
```

Comment:
there's actually at least a minor issue here - if you import the same files multiple times, each time with the same `--override-crs`, this will import all tiles again each time, which is unnecessary.

I'm wondering if we should be checking the CRS of the previous tile itself here (or we could check the CRS of the previous dataset and assume it matches the previous tile, which is *much* cheaper esp if the tiles are on LFS). If the previous dataset/tile matches the override, we can possibly return True here (if the sourceOID matches)

Reply:
fixed this in the last commit.